### PR TITLE
igniton gazebo fullscreen 

### LINKED ISSUE #224: Graphical interface full screen not working
`<gui fullscreen="0">`. The world always launch without fullscreen whether the value is `0` or `1`

### PR DESCRIPTION
**Pull request by mohamedsayed18**
Introduced a new parameter in the scene3D plugin to launch in fullscreen
fix #224

### CODE REVIEW

**Comment by mohamedsayed18** [comment]
`<fullscreen>true</fullscreen>`
The sdf for loading the plugin look like this
```xml
      <plugin filename="GzScene3D" name="3D View">
        <ignition-gui>
          <title>3D View</title>
          <property type="bool" key="showTitleBar">false</property>
          <property type="string" key="state">docked</property>
        </ignition-gui>

        <engine>ogre2</engine>
        <scene>scene</scene>
        <ambient_light>0.4 0.4 0.4</ambient_light>
        <background_color>0.8 0.8 0.8</background_color>
        <fullscreen>true</fullscreen>
      </plugin>
```

**Comment by chapulina** [comment]
@mohamedsayed18 is working on adding support for toggling fullscreen with the F11 key

**Comment by mohamedsayed18** [comment]
the F11 now toggle between the fullscreen and normal window, I tried to make it maximized window but it didn't work I think maybe it is not supported for qwindow

File: src/gui/plugins/scene3d/Scene3D.cc [hunk 1]
```diff
@@ -777,6 +777,23 @@ void IgnRenderer::HandleKeyPress(QKeyEvent *_e)
...
 
+  // fullscreen
+  if (_e->key() == Qt::Key_F11)
+  {
+    if (ignition::gui::App()->findChild
+        <ignition::gui::MainWindow *>()->QuickWindow()->visibility() 
+        == QWindow::FullScreen)
+    {
+      ignition::gui::App()->findChild
+        <ignition::gui::MainWindow *>()->QuickWindow()->showMaximized());
```
Comment (chapulina) [inline src/gui/plugins/scene3d/Scene3D.cc:788]: Oops, this is not compiling, need to remove the extra `)`:

```suggestion
        <ignition::gui::MainWindow *>()->QuickWindow()->showMaximized();
```

Comment (mohamedsayed18) [reply src/gui/plugins/scene3d/Scene3D.cc:788]: Maybe I made a mistake while merging or so.

**Comment by mohamedsayed18** [comment]
It is working and compiling right now

File: src/gui/plugins/scene3d/Scene3D.cc [hunk 2]
```diff
@@ -777,6 +777,23 @@ void IgnRenderer::HandleKeyPress(QKeyEvent *_e)
...
 
+  // fullscreen
+  if (_e->key() == Qt::Key_F11)
+  {
+    if (ignition::gui::App()->findChild
+        <ignition::gui::MainWindow *>()->QuickWindow()->visibility() 
+        == QWindow::FullScreen)
+    {
+      ignition::gui::App()->findChild
+        <ignition::gui::MainWindow *>()->QuickWindow()->showMaximized();
```
Comment (chapulina) [inline src/gui/plugins/scene3d/Scene3D.cc:788]: How about using `showNormal` here instead? For me, that's bringing the window to the state it had before going fullscreen.

```suggestion
        <ignition::gui::MainWindow *>()->QuickWindow()->showNormal();
```

**Comment by mohamedsayed18** [comment]
* The world at the start
fullscreen1

* After pressing F11 (Fullscreen mode)
Screenshot from 2020-08-14 18-11-06

* After pressing F11 (Normal mode)
full2

**Pull request merged**